be/src/storage/rowset/segment.cpp

```diff
@@ -82,9 +82,14 @@ StatusOr<std::shared_ptr<Segment>> Segment::open(std::shared_ptr<FileSystem> fs,
 }
 
 Status Segment::parse_segment_footer(RandomAccessFile* read_file, SegmentFooterPB* footer, size_t* footer_length_hint,
-                                     const FooterPointerPB* partial_rowset_footer) {
+                                     const FooterPointerPB* partial_rowset_footer, uint64_t segment_size) {
     // Footer := SegmentFooterPB, FooterPBSize(4), FooterPBChecksum(4), MagicNumber(4)
-    ASSIGN_OR_RETURN(auto file_size, read_file->get_size());
+    uint64_t file_size;
+    if (segment_size != 0) {
+        file_size = segment_size;
+    } else {
```

Comment:
This is unnecessary, because if we can know the segment file size, we should have opened the `read_file` with a `FileInfo` and `read_file` would have saved the file size in its internal state, which means `read_file->get_size()` is a cheap operation.